[HIPIFY][doc] LLVM 18.1.5 is the latest supported LLVM release

emankov opened the pull request:
+ No patches are needed
+ Updated the documentation accordingly
+ Tested on Windows 11 and Ubuntu 23.10

Rmalavally reviewed (commented):
Looks good, Evgeniy.

emankov committed:
[HIPIFY][doc] LLVM 18.1.5 is the latest supported LLVM release

+ No patches are needed
+ Updated the `README.md` accordingly
+ Tested on Windows 11 and Ubuntu 23.10
